icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -422,10 +422,11 @@ class Parser {
                     break;
 
                 case COMPOUND_PART_TIMES:
+                    if (fAfterPer) result.dimensionality = -1;
```

Comment:
I've not yet found an ICU style guide - where is the guide we should follow? This has impact on the .clang-format config Younies and I have been using - I want to update that to what makes sense. (I found various examples of one-liners, but I suppose it might be old code / not following a style guide.) Current .clang-format we're using will keeps the braces if you add them, but put it all on one line.

Reply:
https://sites.google.com/site/icuprojectuserguide/dev/codingguidelines

Reply:
Adding braces is requested by the style guide. This is easy:

    if (fAfterPer) { result.dimensionality = -1; }

Splitting it onto three lines requires either disagreeing with clang-format in this case, or setting:

    AllowShortIfStatementsOnASingleLine: false

- but then that's at odds with what I see is very common in ICU code, including very new ICU code:

    if (foo) { return; }

In the case of measunit_extra.cpp, "foo" is only "U_FAILURE(status)", which one _could_ argue is a special case.

Just trying to decide on appropriate style guidance for my future code, for clang-format options (which of course aren't flexible enough for very special cases), and for the style guide itself: should it be clarified?

I did find a reference to a rule: "even if that means bending the single-statement-per-line rule slightly" - though I didn't find such a rule itself. (A fully exhaustive reading of the whole page is still on my to-do list.)

Reply:
There are enough developers who like to put short if statements on one line that at this point I think it's accepted ICU style.